cf-agent/files_editxml.c

```diff
@@ -382,8 +384,10 @@ static bool VerifyXPathBuild(EvalContext *ctx, Attributes a, const Promise *pp,
 
 /***************************************************************************/
 
-static PromiseResult VerifyTreeDeletions(EvalContext *ctx, Attributes a, const Promise *pp, EditContext *edcontext)
+static PromiseResult VerifyTreeDeletions(EvalContext *ctx, const Attributes *attr, const Promise *pp, EditContext *edcontext)
 {
+    assert(attr != NULL);
+    Attributes a = *attr; // TODO: Remove this copy
     xmlDocPtr doc = NULL;
```

Comment:
Right. Because this code modifies the copy but the current behavior is to not modify the original Attributes instance held by the caller. :heavy_check_mark:

Reply:
Yes, I'm preserving the current behavior, even though it looks stupid. Incremental improvements ;)